pkg/api/validation/schema.go

```diff
@@ -147,8 +148,15 @@ func (s *SwaggerSchema) ValidateObject(obj interface{}, apiVersion, fieldName, t
 	return allErrs
 }
 
+var versionRegexp = regexp.MustCompile(`^v.+\..*`)
+
 func (s *SwaggerSchema) validateField(value interface{}, apiVersion, fieldName, fieldType string, fieldDetails *swagger.ModelProperty) errs.ValidationErrorList {
-	if strings.HasPrefix(fieldType, apiVersion) {
+	// TODO: caesarxuchao: because we have multiple group/versions and objects
+	// may reference objects in other group, the commented out way of checking
+	// if a filedType is a type defined by us is outdated. We use a hacky way
+	// for now.
```

Comment:
It's because an experimental type may have a v1 type as a field, so `strings.HasPrefix(fieldType, apiVersion)` returns false when it should return true.